Sources/StreamChatUI/Utils/AudioPlayer/AudioPlaying.swift

```diff
@@ -0,0 +1,383 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Foundation
+import StreamChat
+
+/// A protocol describing an object that can be manage the playback of an audio file or stream.
+public protocol AudioPlaying: AnyObject {
+    /// Provides  way to get an instance of a player
+    static func build() -> AudioPlaying
+
+    /// Requests the playbackContext for the given URL. If the player's current item has as URL that
+    /// matches the provided one, it should return a context, otherwise it will return
+    /// ``AudioPlaybackContext.notLoaded``
+    /// - Parameters:
+    /// - url: The URL (provided by the asset) that is used to stream/download the content to play
+    func playbackContext(for url: URL) -> AudioPlaybackContext
+
+    /// Instructs the player to load the asset from the provided URL and prepare it for streaming. If the
+    /// player's current item has as URL that matches the provided one, then we will try to play or restart
+    /// the playback while updating the new delegate.
+    /// - Parameters:
+    /// - url: The URL where the asset will be streamed from. If nil then the player will simply clear
+    /// up the current playback queue.
+    /// - delegate: The delegate that will be informed for changes on the asset's playback.
+    func loadAsset(from url: URL?, andConnectDelegate delegate: AudioPlayingDelegate)
```

Comment:
As we are using a single player for everything, if we were to introduce multicast delegates we will need to add more logic to the view itself. For now (with the requirements we have) this doesn't seem to be necessary. Additionally, this player, is only going to be used for audio messages playback which don't really have many requirements (e.g. fast forward, rewind or even previews). For this reason I'm not sure there are going to be additional events added here. If there are, we can handle those in future itereations.
(Please note that even though we are using an AVPlayer here (the same that is being used for Video playback) our use cases are much simpler than video playback).

Reply:
I think it will be pretty common for a customer to want to observe the audio player events in a different UI component or any other component. If we add the multicastDelegate later, then we could potentially deprecate the API quickly since the delegate won't make sense. Either way, I'll leave it up to you 👍 If we can change this later whenever a customer needs it is also fine 👍

Reply:
I can't really see now (as there are not use cases) on what kind of other events customers may want to subscribe to. As this is currently only implemented for the playback of audiomessages and not audio, I would prefer to keep small and simple and closer to the defined needs rather than trying to guess potential usecases.